crates/node/src/test_utils.rs

```diff
@@ -23,6 +25,54 @@ pub struct TestGenerators {
 
 type ParticipantAndProtocol<T> = (Participant, Box<dyn Protocol<Output = T>>);
 
+// +++++++++++++++++ Any Protocol +++++++++++++++++ //
+/// Run a protocol to completion, synchronously.
+///
+/// This works by executing each participant in order.
+///
+/// The reason this function exists is as a convenient testing utility.
+/// In practice each protocol participant is likely running on a different machine,
+/// and so orchestrating the protocol would happen differently.
+pub fn run_protocol<T>(
```

Comment:
Adding this was also required for an unintended API break :| We thought the function was not being used